src/app/(organization)/projects/[project]/edit/EditProject.tsx

```diff
@@ -1,32 +1,100 @@
+import clsx from 'clsx';
 import { useRouter } from 'next/navigation';
-import { useForm } from 'react-hook-form';
+import { useEffect } from 'react';
+import Dropzone from 'react-dropzone';
+import { Controller, useForm } from 'react-hook-form';
 import Card from '../../../../../components/Card';
+import Divider from '../../../../../components/Divider';
 import Typography, { Size } from '../../../../../components/Typography';
-import { CreateProjectInput, Project } from '../../../../../graphql.types';
+import { EditProjectInput, EditProjectPayload, Project } from '../../../../../graphql.types';
+import { downloadFromUrl } from '../../../../../modules/downloadFile';
+import { uploadFile } from '../../../../../modules/upload';
+import { GetProject } from '../../../../../queries/project.graphql';
+import { EditProject as EditProjectMutation } from './../../../../../mutations/project.graphql';
 
-export default function EditProject({ project }: { project: Project }) {
+interface GetProjectData {
+  project: Project;
+}
+
+interface GetProjectVars {
+  project: string;
+}
+
+interface EditProjectVars {
+  input: EditProjectInput;
+}
+
+interface EditProjectData {
+  editProject: EditProjectPayload;
+}
+interface EditProjectForm {
+  name: string;
+  file: File;
+}
+
+export default function EditProject({ project }: { project: string }) {
   const router = useRouter();
 
-  const { handleSubmit, register } = useForm<CreateProjectInput>({
-    defaultValues: {
-      name: project.name,
-      organization: project.organizationId,
-    },
+  const { control, register, handleSubmit, formState, setValue, reset } =
+    useForm<EditProjectForm>();
+
+  const projectQuery = useQuery<GetProjectData, GetProjectVars>(GetProject, {
+    variables: { project },
   });
-  // TODO: Add edit project mutation
-  const loading = false;
 
-  const submit = async () => {};
+  const projectData = projectQuery.data?.project;
 
   const onClose = () => {
-    if (loading) {
-      return;
+    router.push('/projects');
+  };
+
+  const [editProject, editProjectResult] = useMutation<EditProjectData, EditProjectVars>(
+    EditProjectMutation
+  );
+
+  const loading = editProjectResult.loading;
+
+  const submit = async ({ name, file }: EditProjectForm) => {
+    let profileImageUrl;
+    if (file) {
+      const { url } = await uploadFile(file);
+      profileImageUrl = url;
     }
 
-    router.push('/projects');
+    editProject({
+      variables: {
+        input: {
+          id: project,
+          name,
+          profileImageUrl,
+        },
+      },
+      onCompleted: () => {
+        router.push('/projects');
+      },
+    });
   };
+
+  useEffect(() => {
+    const fetchFile = async (url: string) => {
+      const file = await downloadFromUrl(url);
+      reset({
+        file,
+      });
+    };
+    if (projectData) {
+      if (projectData.profileImageUrl) {
+        fetchFile(projectData.profileImageUrl);
+      }
```

Comment:
The preview should take a string or File if its a string then render set the src as is when file use current method of creating url from the file object.